lib/src/flutter_search_bar_base.dart

```diff
@@ -132,7 +136,18 @@ class SearchBar {
   ///
   AppBar buildSearchBar(BuildContext context) {
     ThemeData theme = Theme.of(context);
-    Color buttonColor = inBar ? null : theme.iconTheme.color;
+    TextStyle hintStyle = TextStyle(
+      color: inBar
+          ? theme.appBarTheme?.textTheme?.headline4?.color ?? theme.primaryTextTheme?.headline4?.color
+          : theme.textTheme.headline4.color,
+    );
+    TextStyle inputStyle = inBar ? theme.appBarTheme?.textTheme?.subtitle1 ?? theme.primaryTextTheme?.subtitle1 : null;
+    Color buttonColor = inBar ? inputStyle?.color : theme.iconTheme.color;
+
+    // This is a bit of a hack, but follows the default initialization of disabled color in a material theme.
+    // If we're in the app bar, we want to invert the default disabled color, unless one is provided.
+    Color disabledColor = this.disabledColor ??
+        (inBar ? (theme.disabledColor == Colors.black38 ? Colors.white38 : Colors.black38) : theme.disabledColor);
```

Comment:
I wonder if it would be a bad idea to do a straight inversion (mathematically) on the color instead of trying to guess what the opposite is for a given value. Thoughts?

Reply:
All the changes I made are based on the Material definition:

```
disabledColor ??= isDark ? Colors.white38 : Colors.black38;
```

Having a mathematical inversion could be the safest pick. Though, the Material theme has plenty of hard-coded entries, which are dictated by the definition of Material design, and will hardly change anytime soon, so in this specific case, I would worry about this whenever a new breaking Material spec comes up?

Reply:
Well, material is gradually evolving to be less of a spec and more of a framework at this point. Google has basically said with Material 2.0 "do whatever you want as long as it is consistent and kind of uses these components"... but I see your point.

This is probably fine for now, especially since the dev can easily provide their own disabledColor if they so choose.